src/topology/algebra/order/left_right.lean

```diff
@@ -61,3 +86,256 @@ lemma continuous_at_iff_continuous_left'_right' {a : α} {f : α → β} :
   continuous_at f a ↔ continuous_within_at f (Iio a) a ∧ continuous_within_at f (Ioi a) a :=
 by rw [continuous_within_at_Ioi_iff_Ici, continuous_within_at_Iio_iff_Iic,
   continuous_at_iff_continuous_left_right]
+
+end topological_space
+
+section left_right_lim
+
+section
+
+variables {α β : Type*} [linear_order α] [topological_space β]
+
+/-- Let `f : α → β` be a function from a linear order `α` to a topological_space `β`, and
+let `a : α`. The limit strictly to the left of `f` at `a`, denoted with `left_lim f a`, is defined
+by using the order topology on `α`. If `a` is isolated to its left, we use `f a` instead to
+guarantee a good behavior in most cases. -/
+@[irreducible] noncomputable def left_lim (f : α → β) (a : α) : β :=
```

Comment:
Does it make sense to put this in the `function` namespace? So that we can still write `f.left_lim a`?

Reply:
Good idea. Done.